File: 2015/01/solution.js
let floor = 0
let basementHit = false
let input = require('fs').readFileSync('input.txt', 'utf8')
for (let i = 0; i < input.length; i++) {
  switch(input.charAt(i)) {
    case '(':
      floor += 1
      break
    case ')':
      floor -= 1
      break
  }
  if (floor < 0 && !basementHit) {
    console.log(`Part 2: Santa first hit the basement at position ${i + 1}`)
    basementHit = true
  }
}

console.log(`Part 1: Santa ends on floor ${floor}`)

File: 2015/01/solution.rb
floor = 0
i = 0
basement_hit = nil
File.read('input.txt').each_char do |char|
  if char == '('
    floor += 1
  elsif char == ')'
    floor -= 1
  end
  i += 1
  if floor < 0 && !basement_hit
    basement_hit = i
  end  
end

puts 'Part 1', floor
puts 'Part 2', basement_hit
